pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -243,6 +244,15 @@ func getHostname(tplVar []byte, svc listeners.Service) ([]byte, error) {
 	return []byte(name), nil
 }
 
+// getExtra returns specific data
+func getExtra(tplVar []byte, svc listeners.Service) ([]byte, error) {
+	value, err := svc.GetExtraConfig(string(tplVar))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get extra info for service %s, skipping config - %s", svc.GetEntity(), err)
+	}
+	return []byte(value), nil
+}
+
```

Comment:
When I see how you are using the `GetExtraConfig(...)` method, I'm wondering if we can simplify it by having a method return directly a generic interface. Then it is up to you do cast it in the correct type when you process the `Service`
```go
getExtraConfig() interface{}
```
or like the decoder method.

```golang
getExtraConfig(cfg interface{}) error
//...
var snmpServiceConfig SNMPServiceConfig
err := svc.getExtraConfig(&snmpServiceConfig)
```

Reply:
+1
this could be useful for future listeners

Reply:
@clamoriniere Can you explain a bit more or give an example where it's done? I don't get how you get from your suggestion to actual bytes in the resolver.

Reply:
I tried to implement my idea: https://play.golang.org/p/RFBZIJN5NKf

Reply:
OK so what I don't get is how you can pass service specific data to `GetExtraConfig`? And also handle service specific data in the response? At that point I don't know what I'm handling, I have a "generic" service right?

Reply:
Sorry I didn't understand how `func getExtra()` was used. So forget about my previous comment

just a small comment: the function can use `[]byte` instead of `string` since we will always cast it after.
```golang
GetExtraConfig([]byte) ([]byte, error)
//or
GetExtraConfig(string) ([]byte, error)
```